packages/design-system/src/components/Drawer/Drawer.tsx

```diff
@@ -69,55 +69,77 @@ export interface DrawerProps {
  * [refer to its full documentation page](https://design.cms.gov/components/drawer/).
  */
 export const Drawer = (props: DrawerProps) => {
-  const headingRef = useRef(null);
-  const headingId = useId('drawer--', props.headingId);
+  const {
+    ariaLabel,
+    backdropClickExits,
+    children,
+    className,
+    closeButtonText,
+    closeButtonVariation,
+    footerBody,
+    footerTitle,
+    hasFocusTrap,
+    heading,
+    headingId,
```

Comment:
One pattern we have in a couple components right now to solve competing prop names and variable names is redefining the prop with a prefix of `user*`, like `headingId: userHeadingId`. That might prevent us from accidentally using `headingId` later on when we mean `headingElId`. See Dropdown's `userOnBlur` and Autocomplete's `userInputRef`.